segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/StorageTestBase.java

```diff
@@ -86,6 +86,87 @@ public void testCreate() {
         }
     }
 
+    /**
+     * Tests the exists API.
+     */
+    @Test
+    public void testListSegmentsWithOneSegment() {
+        String segmentName = "foo_open";
+        try (Storage s = createStorage()) {
+            s.initialize(DEFAULT_EPOCH);
+
+            Iterator<SegmentProperties> iterator = s.listSegments().join();
+            Assert.assertFalse(iterator.hasNext());
+
+            createSegment(segmentName, s);
+
+            iterator = s.listSegments().join();
+            Assert.assertTrue(iterator.hasNext());
+            SegmentProperties prop = iterator.next();
+            Assert.assertEquals(prop.getName(), segmentName);
+            Assert.assertFalse(iterator.hasNext());
+        }
+    }
+    /**
+     * Tests the ability to list Segments.
+     */
+    @Test
+    public void testListSegments() throws Exception {
+        @Cleanup
+        val baseStorage = new InMemoryStorage();
+        @Cleanup
+        val s = new RollingStorage(baseStorage, new SegmentRollingPolicy(1));
+        Set<String> sealedSegments = new HashSet<>();
+        s.initialize(1);
+        int expectedCount = 50;
+        byte[] data = "data".getBytes();
+        for (int i = 0; i < expectedCount; i++) {
+            String segmentName = "segment-" + i;
+            val wh1 = s.create(segmentName);
+            // Write data.
+            s.write(wh1, 0, new ByteArrayInputStream(data), data.length);
+            if (rand.nextInt(2) == 1) {
+                s.seal(wh1);
+                sealedSegments.add(segmentName);
+            }
+        }
+        Iterator<SegmentProperties> it = s.listSegments();
+        int actualCount = 0;
+        while (it.hasNext()) {
+            SegmentProperties curr = it.next();
+            //check the length matches
+            Assert.assertEquals(curr.getLength(), data.length);
+            if (sealedSegments.contains(curr.getName()))
+                Assert.assertTrue(curr.isSealed());
+            else
+                Assert.assertFalse(curr.isSealed());
+            ++actualCount;
+        }
+        Assert.assertEquals(actualCount, expectedCount);
+    }
+    @Test
+    public void testListSegmentsWithDeletes() throws Exception{
+        @Cleanup
+        val baseStorage = new InMemoryStorage();
+        @Cleanup
+        val s = new RollingStorage(baseStorage, new SegmentRollingPolicy(1));
+        s.initialize(DEFAULT_EPOCH);
+        Set<String> deletedSegments = new HashSet<>();
+        int expectedCount = 50;
+        for(int i=0; i < expectedCount; i++) {
+            String segmentName = "segment-"+i;
+            SegmentHandle handle = s.create(segmentName);
+            if(rand.nextInt(2) == 1) {
+                s.delete(handle);
+                deletedSegments.add(segmentName);
+            }
+        }
+
+        Iterator<SegmentProperties> it = s.listSegments();
+        expectedCount -= deletedSegments.size();
+        Assert.assertEquals(expectedCount, Iterators.size(it));
```

Comment:
Should we check contents of the list returned to make sure deleted files are not present in that list ? (and all that were expected are present)